src/parser/priest/holy/modules/talents/75/PrayerCircle.tsx

```diff
@@ -0,0 +1,84 @@
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
+import SPELLS from 'common/SPELLS';
+import React from 'react';
+import Events, { BeginCastEvent, CastEvent } from 'parser/core/Events';
+import Statistic from 'interface/statistics/Statistic';
+import STATISTIC_CATEGORY from 'interface/others/STATISTIC_CATEGORY';
+import BoringSpellValueText from 'interface/statistics/components/BoringSpellValueText';
+import STATISTIC_ORDER from 'interface/others/STATISTIC_ORDER';
+import AbilityTracker from 'parser/priest/holy/modules/core/AbilityTracker';
+import Abilities from 'parser/priest/holy/modules/Abilities';
+
+const PR_BUFF_DURATION = 8000;
+
+/*
+  Using Circle of Healing reduces the cast time of your Prayer of Healing by 25% for 8 sec.
+*/
+class PrayerCircle extends Analyzer {
+  static dependencies = {
+    abilityTracker: AbilityTracker,
+    abilities: Abilities,
+  };
+  protected abilityTracker!: AbilityTracker;
+  protected abilities!: Abilities;
+
+  lastCohCastAt: number = 0;
+  lastPohCastStart: {
+    timestamp: number,
+    isBuffed: boolean,
+    expectedCooldown: number,
+  } = { timestamp: 0, isBuffed: false, expectedCooldown: 0 };
+  buffedCohCasts = 0;
+
+  get unbuffedCohCasts() {
+    return (this.abilityTracker.getAbility(SPELLS.PRAYER_OF_HEALING.id).casts || 0) - this.buffedCohCasts;
+  }
+
+  constructor(options: any) {
+    super(options);
+    this.active = true; // this.selectedCombatant.hasTalent(SPELLS.PRAYER_CIRCLE_TALENT.id);
+
+    if (this.active) {
+      this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.CIRCLE_OF_HEALING_TALENT), this.cohCast);
+      this.addEventListener(Events.begincast.by(SELECTED_PLAYER).spell(SPELLS.PRAYER_OF_HEALING), this.startPohCast);
+      this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.PRAYER_OF_HEALING), this.finishPohCast);
+    }
+  }
+
+  cohCast(event: CastEvent) {
+    this.lastCohCastAt = event.timestamp;
+  }
+
+  startPohCast(event: BeginCastEvent) {
+    this.lastPohCastStart.timestamp = event.timestamp;
```

Comment:
I could be very wrong but in my head I was picturing that this would work:
```typescript
lastCastStartWasBuffed: boolean;

startPohCast(event: BeginCastEvent) {
    // entire comment revolves around this buff existing; I dont see it referenced anywhere in here so maybe it doesnt?
    this.lastCastStartWasBuffed = this.selectedCombatant.hasBuff(SPELLS.PRAYER_CIRCLE_BUFF.id); 
}

finishPohCast(event: CastEvent) {
  if (this.lastCastStartWasBuffed) {
    this.buffedCohCasts += 1;
  }
}
```

The logic being that the boolean value will always be the most "recent" begincast event for PoH, so you can use that when the cast finishes to see if the buff was active when the cast started

Reply:
I don't know the ID of the buff, so I figured that this was good enough for now.

Reply:
Found a log with the talent:

https://wowanalyzer.com/report/y3pDAt2R1HLjdBNC/7-Heroic+Kael'thas+-+Wipe+7+(3:36)/Myixi/standard/events

321379

But I'll leave it up to you what to do with it for now. Everything LGTM for you to merge if you'd rather leave it for later

Reply:
updated to use the buff! Where do you get logs that have shadowlands content? I asked in discord, but I never got an answer :D

Reply:
Sharqq pointed out that at the very bottom of Warcraftlogs front page you can now see beta logs and search the same way you do with live logs. I didnt find out until today either XD